src/helm/benchmark/compute_contamination_metrics.py

```diff
@@ -0,0 +1,212 @@
+        self._input_bits |= metric._input_bits
+        self._reference_bits |= metric._reference_bits
+
+    @property
+    def num_input_positive_instances(self):
+        return self._input_bits.count()
+
+    @property
+    def num_reference_positive_instances(self):
+        return self._reference_bits.count()
+
+    @property
+    def input_positive_rate(self):
+        return self._input_bits.count() / self.num_instances
+
+    @property
+    def reference_positive_rate(self):
+        return self._reference_bits.count() / self.num_instances
+
+    def generate_summary(self, tags) -> str:
+        """Output a summary of the metric"""
+        summary = {
+            "Setting": f"{self.scenario_name},{','.join(tags)}",
+            "total_instances": self.num_instances,
+            "num_input_positive_instances": self.num_input_positive_instances,
+            "num_reference_positive_instances": self.num_reference_positive_instances,
+            "input_positive_rate": self.input_positive_rate,
+            "reference_positive_rate": self.reference_positive_rate,
+        }
+        return json.dumps(summary)
+
+
+def load_scenarios_from_jsonl(filename: str) -> List[LightScenario]:
+    """
+    Create a list of light scenarios from a jsonl file.
+
+    Input File Format:
+
+    Instance JSON 1
+    Instance JSON 2
+    Instance JSON 3
+    ...
+
+    Each line is a json and each json looks like:
+    {
+        "name": "SCENARIO_NAME",
+        "light_instances": [
+            {
+            "input": "INPUT_TEXT1",
+            "references": [
+                "REFERENCE_TEXT_1",
+                "REFERENCE_TEXT_2"
+            ]
+            },
+            {
+            "input": "INPUT_TEXT2",
+            "references": [
+                "REFERENCE_TEXT_3",
+                "REFERENCE_TEXT_4"
+            ]
+            }
+        ]
+    }
+    """
+
+    def create_light_instance_from_dict(instance_dict: dict) -> LightInstance:
+        return LightInstance(input=instance_dict["input"], references=instance_dict["references"])
+
+    light_scenarios: List[LightScenario] = []
+    scenario_jsons = open(filename, "r").readlines()
+    for scenario_json in scenario_jsons:
+        scenario_dict: dict = json.loads(scenario_json)
+        scenario_name: str = scenario_dict["name"]
+        light_instances: List[LightInstance] = [
+            create_light_instance_from_dict(instance_dict) for instance_dict in scenario_dict["light_instances"]
+        ]
+        light_scenarios.append(LightScenario(name=scenario_name, light_instances=light_instances))
+    return light_scenarios
+
+
+def compute_instance_contamination(scenario: LightScenario, training_document: str, n: int) -> BinaryScenarioMetric:
+    def get_ngrams(text: str):
+        return list(ngrams(text.split(), n))
+
+    scenario_metric = BinaryScenarioMetric.from_scenario(scenario)
+    document_ngrams = set(get_ngrams(training_document))
```

Comment:
For your test runs, what are how many document ngrams are there and how many scenario ngrams are there?

For the final run, how many scenario ngrams are we expecting (do they all fit in memory?